flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -111,6 +111,12 @@ public FlinkCatalog(
 
   @Override
   public void open() throws CatalogException {
+    // Create the default database if it does not exist.
+    try {
+      createDatabase(getDefaultDatabase(), ImmutableMap.of(), true);
+    } catch (DatabaseAlreadyExistException e) {
+      throw new CatalogException(e);
```

Comment:
Does this do what you want? It seems like if the intent is `IF NOT EXISTS` then you should ignore the `DatabaseAlreadyExistException`.

Reply:
Okay, I see that it delegates. When would this throw `DatabaseAlreadyExistsException` then? Is this just to ignore a checked exception?

Reply:
Yeah, it delegates the `DatabaseAlreadyExistsException`, in theory it should not throw this exception.  I noticed this before, we could just ignore it here.